File: 2279. Maximum Bags With Full Capacity of Rocks/main.js
/**
 * @param {number[]} capacity
 * @param {number[]} rocks
 * @param {number} additionalRocks
 * @return {number}
 */
var maximumBags = function (capacity, rocks, a) {
    let res = 0;
    let arr = [];

    for (let i = 0; i < rocks.length; i++) {
        arr.push(capacity[i] - rocks[i]);
    }

    arr.sort((a, b) => a - b);

    for (let i = 0; i < arr.length; i++) {
        if (a <= 0 || arr[i] - a > 0) return res;

        a -= arr[i];
        res++;
    }

    return res;
};

File: 2279. Maximum Bags With Full Capacity of Rocks/main.py
class Solution:
    def maximumBags(self, capacity: List[int], rocks: List[int], a: int) -> int:
        res = 0
        arr = []

        for i in range(len(capacity)):
            arr.append(capacity[i]-rocks[i])
        
        arr.sort()
        
        for i in range(len(arr)):
            if a < 1 or arr[i] - a > 0:
                return res
            a -= arr[i]
            res += 1

        return res